deepchem/molnet/load_function/pdbbind_datasets.py

```diff
@@ -1,458 +1,124 @@
-  """Featurizes pdbbind according to provided featurization"""
-  tasks = ["-logKd/Ki"]
-  data_dir = deepchem.utils.data_utils.get_data_dir()
-  pdbbind_dir = os.path.join(data_dir, "pdbbind")
-  dataset_dir = os.path.join(pdbbind_dir, "%s_%s" % (subset, feat))
-
-  if not os.path.exists(dataset_dir):
-    deepchem.utils.data_utils.download_url(
-        "https://deepchemdata.s3-us-west-1.amazonaws.com/featurized_datasets/core_grid.tar.gz"
-    )
-    deepchem.utils.data_utils.download_url(
-        "https://deepchemdata.s3-us-west-1.amazonaws.com/featurized_datasets/full_grid.tar.gz"
-    )
-    deepchem.utils.data_utils.download_url(
-        "https://deepchemdata.s3-us-west-1.amazonaws.com/featurized_datasets/refined_grid.tar.gz"
-    )
-    if not os.path.exists(pdbbind_dir):
-      os.system('mkdir ' + pdbbind_dir)
-    deepchem.utils.data_utils.untargz_file(
-        os.path.join(data_dir, 'core_grid.tar.gz'), pdbbind_dir)
-    deepchem.utils.data_utils.untargz_file(
-        os.path.join(data_dir, 'full_grid.tar.gz'), pdbbind_dir)
-    deepchem.utils.data_utils.untargz_file(
-        os.path.join(data_dir, 'refined_grid.tar.gz'), pdbbind_dir)
-
-  return deepchem.data.DiskDataset(dataset_dir), tasks
 
+import deepchem as dc
+from deepchem.molnet.load_function.molnet_loader import TransformerGenerator, _MolnetLoader
+from deepchem.data import Dataset
+from typing import List, Optional, Tuple, Union
 
-def load_pdbbind_grid(split="random",
-                      featurizer="grid",
-                      subset="core",
-                      reload=True):
-  """Load PDBBind datasets. Does not do train/test split"""
-  if featurizer == 'grid':
-    dataset, tasks = featurize_pdbbind(feat=featurizer, subset=subset)
+PDBBIND_URL = "https://deepchemdata.s3-us-west-1.amazonaws.com/datasets/pdbbindv2019/pdbbind_v2019_refined.tar.gz"
+PDBBIND_TASKS = ['-logKd/Ki']
 
-    splitters = {
-        'index': deepchem.splits.IndexSplitter(),
-        'random': deepchem.splits.RandomSplitter(),
-        'time': deepchem.splits.TimeSplitterPDBbind(dataset.ids)
-    }
-    splitter = splitters[split]
-    train, valid, test = splitter.train_valid_test_split(dataset)
 
-    transformers = []
-    for transformer in transformers:
-      train = transformer.transform(train)
-    for transformer in transformers:
-      valid = transformer.transform(valid)
-    for transformer in transformers:
-      test = transformer.transform(test)
-
-    all_dataset = (train, valid, test)
-    return tasks, all_dataset, transformers
-
-  else:
-    data_dir = deepchem.utils.data_utils.get_data_dir()
-    if reload:
-      save_dir = os.path.join(
-          data_dir, "pdbbind_" + subset + "/" + featurizer + "/" + str(split))
-
-    dataset_file = os.path.join(data_dir, subset + "_smiles_labels.csv")
+class _PDBBindLoader(_MolnetLoader):
 
+  def create_dataset(self) -> Dataset:
+    dataset_file = os.path.join(self.data_dir, "pdbbind_v2019_refined.tar.gz")
```

Comment:
Could we add options to get the full set as well? Might be useful for benchmarking methods at larger scale